pkg/kubelet/network/dns/dns.go

```diff
@@ -98,19 +108,49 @@ func (c *Configurer) getClusterDNS(pod *v1.Pod) ([]net.IP, error) {
 		networkName = arktosv1.NetworkDefault
 	}
 
-	if network, err := c.arktosV1Client.NetworksWithMultiTenancy(pod.Tenant).Get(networkName, metav1.GetOptions{}); err == nil {
-		if network.Status.Phase != arktosv1.NetworkReady {
-			return nil, fmt.Errorf("network %s/%s is in %q phase, not ready", network.Tenant, network.Name, network.Status.Phase)
-		}
+	if c.serviceListers == nil || len(c.serviceListers) < 1 {
+                return nil, fmt.Errorf("failed to get serviceListers when getting service IP for network")
+	}
 
-		ip := net.ParseIP(network.Status.DNSServiceIP)
-		if ip != nil {
-			return []net.IP{ip}, nil
-		}
+	var index int
+	if len(c.serviceListers) > 1 {
+		klog.V(4).Infof("Need locate which serviceLister is correct one in multi-servceListers")
+		// To get correct index of serviceLister based on Pod's tenant information
+		index = c.clientManager.PickClient(pod.Tenant);
 	} else {
-		klog.Errorf("failed to get network %s/%s: %v", pod.Tenant, networkName, err)
+		// Only has one serviceLister
+		index = 0;
+	}
+	klog.V(4).Infof("Pod NAME: %q | Index : %q | networkName : %q | Tenant : %q ", pod.Name, index,  networkName, pod.Tenant)
+	services, err := c.serviceListers[index].List(labels.Everything())
```

Comment:
the logic in DNS below ( in the for loop ) only needs the dns service name, so it probably better just use GET to get this service from the api server instead of a LIST